src/charm.py

```diff
@@ -962,7 +1080,9 @@ def _restart_traefik(self):
 
         if _TRAEFIK_SERVICE_NAME not in current_services:
             self.unit.status = MaintenanceStatus(f"creating the {_TRAEFIK_SERVICE_NAME!r} service")
-            self.container.add_layer(_TRAEFIK_LAYER_NAME, layer, combine=True)
+            self.container.add_layer(
+                _TRAEFIK_LAYER_NAME, typing.cast(LayerDict, layer), combine=True
```

Comment:
Seems wrong that we need the cast here.

Reply:
either here, or at layer instantiation. Pyright doesn't like `foo:LayerDict = {...}` either, for reasons unknown. It's too complex a type. Too many optional fields